### PR TITLE
Add torch-squim

### PR DESCRIPTION
**Pull request by shimhz**

### CODE REVIEW

File: speech_evaluation/utterance_metrics/squim.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    if fs != 16000:
+        gt_x = F.resample(gt_x, fs, 16000)
+        pred_x = F.resample(pred_x, fs, 16000)
+
+    subjective_model = SQUIM_SUBJECTIVE.get_model()
+    objective_model = SQUIM_OBJECTIVE.get_model()
+
+    mos = subjective_model(pred_x, gt_x)
+    stoi, pesq, si_sdr = objective_model(pred_x)
```
Comment (ftshijt) [inline speech_evaluation/utterance_metrics/squim.py]: I think we may skip the three metrics, since they already supported?

Comment (ftshijt) [inline speech_evaluation/utterance_metrics/squim.py]: Is the API correct? I thought mos will only need pred, but others need gt as well

**Review by ftshijt — COMMENTED** [review]
Thanks for the update! Some minor comments as follow

File: speech_evaluation/utterance_metrics/squim.py (hunk 1 above)
Comment (shimhz) [reply speech_evaluation/utterance_metrics/squim.py]: Sure. you mean the objective model, right?

Comment (shimhz) [reply speech_evaluation/utterance_metrics/squim.py]: Yex, I also had doubts at first, but this is correct~

Comment (ftshijt) [reply speech_evaluation/utterance_metrics/squim.py]: Yeah~

File: speech_evaluation/utterance_metrics/squim.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    gt_x = gt_x.float()
+
+    pred_x = torch.from_numpy(pred_x).unsqueeze(0)
+    pred_x = pred_x.float()
+
+    subjective_model = SQUIM_SUBJECTIVE.get_model()
+    mos = subjective_model(pred_x, gt_x)
+
+    #objective_model = SQUIM_OBJECTIVE.get_model()
+    #stoi, pesq, si_sdr = objective_model(pred_x)
```
Comment (ftshijt) [inline speech_evaluation/utterance_metrics/squim.py]: Apology for my misunderstanding, as I go over their methods, they seem to use a pre-trained model to conduct the objective estimation to achieve no-reference prediction. In that case, I think they are also very cool to be included.

File: speech_evaluation/utterance_metrics/squim.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    
+    return {"mos": mos.detach().numpy()}
```
Comment (ftshijt) [inline speech_evaluation/utterance_metrics/squim.py]: Also, their pseudo subjective metric is not in the same setting as the normal mos. On my side, I would suggest to use a specific name for the collected metric instead of "mos, stoi, pesq, si_sdr", which might be confusing to existing known metrics.

In that case, how about simply name them as:
- torch_squim_mos
- torch_squim_stoi
- torch_squim_pesq
- torch_squim_si_sdr
or something similar with clear noting?

File: speech_evaluation/utterance_metrics/squim.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    import torchaudio.functional as F
+    from torchaudio.pipelines import SQUIM_OBJECTIVE, SQUIM_SUBJECTIVE
+except ImportError:
+    raise ImportError(
+        "Please install pesq, pystoi, torchaudio and retry: pip install stoi"
+    )
+
+
+def squim_metric(pred_x, gt_x, fs):
+
```
Comment (ftshijt) [inline speech_evaluation/utterance_metrics/squim.py:29]: For better reference, it would be great if you could add the corresponding reference paper/links as the docstring to this function.

File: egs/codec_16k.yaml [hunk 5]
```diff
@@ -62,6 +62,10 @@
 - name: speaker
   model_tag: default
 
+# torchaudio-squim
+# -- torchaudio-squim: non-intrusive speech assessment in torchaudio
```
Comment (ftshijt) [inline egs/codec_16k.yaml:66]: Probably also good to include some references here?

**Review by ftshijt — COMMENTED** [review]
Many thanks. Apology that I may misunderstand some of the components as I did not carefully check the torch-squim paper at the time. After I recheck their claim, I would suggest some additional modifications as follows (sorry for making it complicated and thank again for your great help to the toolkit!)

**Comment by shimhz** [comment]
Reflected all the comments you gave! Thank you for checking :-)

**Comment by ftshijt** [comment]
Many thanks! I merged the PR~
